### PR TITLE
Web console talks to master over configured master host/port and embedded web console allowed by default in CORS list

### PR DESCRIPTION
**Pull request by jwforres**

### CODE REVIEW

**Comment by jwforres** [comment]
@smarterclayton @liggitt

File: pkg/cmd/server/origin/master.go [hunk 1]
```diff
@@ -295,13 +301,19 @@ func (c *MasterConfig) RunAssetServer() {
...
-						&assetfs.AssetFS{
-							assets.Asset,
-							assets.AssetDir,
-							"",
-						},
+				assets.GeneratedConfigHandler(
+					masterURL.Host,
+					OpenShiftAPIPrefixV1Beta1,
+					masterURL.Host, // when we support separate k8s server this is k8s addr
+					"/api/v1beta1", // when we support separate k8s server this is k8s prefix
```
Comment (jwforres) [inline pkg/cmd/server/origin/master.go]: @smarterclayton is there a better way to get the k8s values here?

**Comment by jwforres** [comment]
@liggitt updated based on clayton's feedback, ready for you to review

File: pkg/cmd/server/start.go [hunk 2]
```diff
@@ -210,10 +213,15 @@ func start(cfg *config, args []string) error {
 
 		assetAddr := net.JoinHostPort(cfg.MasterAddr.Host, strconv.Itoa(cfg.BindAddr.Port+1))
 
+		// always include the all-in-one server's web console as an allowed CORS origin
+		// always include localhost as an allowed CORS origin
+		cfg.CORSAllowedOrigins = append(cfg.CORSAllowedOrigins, assetAddr, "localhost")
```
Comment (liggitt) [inline pkg/cmd/server/start.go]: 127.0.0.1?

Comment (smarterclayton) [inline pkg/cmd/server/start.go]: Dear god yes.

----- Original Message -----

File: assets/app/scripts/services/data.js [hunk 3]
```diff
@@ -483,11 +469,25 @@ angular.module('openshiftConsole')
     }
   };
 
-  var URL_ROOT_TEMPLATE = "{protocol}://{+serverUrl}/{apiRoot}/{apiVersion}/";
+  var URL_ROOT_TEMPLATE = "{protocol}://{+serverUrl}{+apiPrefix}/";
```
Comment (liggitt) [inline assets/app/scripts/services/data.js]: Do we want to be told the api version, or have the UI know the version of the API it was coded against? Seems like the UI should get to stay on a k8s and openshift API version until it is ready to move up

File: pkg/cmd/server/start.go (hunk 2 above)
Comment (wjiangjay) [inline pkg/cmd/server/start.go]: Means can only access from localhost? Is this reasonable?

Comment (jwforres) [inline pkg/cmd/server/start.go]: No it means localhost is added to the list of Origins that will result in CORS headers being returned, necessary for the web console to talk to the API server.  That line adds both localhost and assetAddr where assetAddr is currently defined as whatever you configured for the master's hostname:port + 1 on the port.   This list of allowed origins can always be added to by specifying the --cors-allowed-origins flag during openshift start

**Comment by liggitt** [comment]
LGTM

**Comment by jwforres** [comment]
[merge]

**Comment by jwforres** [comment]
unrelated failure [merge]

**Comment by jwforres** [comment]
and again [merge]

**Comment by jwforres** [comment]
[merge]

**Comment by jwforres** [comment]
try again now that merge issue should be resolved [merge]

**Pull request merged**